compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -73,7 +73,19 @@
 #define FMA_CONST_HIGHBOUNDI2D 8.371160993643E298
 #define FMA_CONST_HIGHBOUNDF2D 5.282945626245E269
 
+// Returns true if signed 64-bit a + b has overflow or underflow
+#define IS_INT64_ADD_OVERFLOW(a, b)  (b > 0 && a > TR::getMaxSigned<TR::Int64>() - b)
+#define IS_INT64_ADD_UNDERFLOW(a, b) (b < 0 && a < TR::getMinSigned<TR::Int64>() - b)
 
+// Returns true if signed 64-bit a - b has overflow or underflow
+#define IS_INT64_SUB_OVERFLOW(a, b)  (b < 0 && a > TR::getMaxSigned<TR::Int64>() + b)
+#define IS_INT64_SUB_UNDERFLOW(a, b) (b > 0 && a < TR::getMinSigned<TR::Int64>() + b)
+
+// Returns true if unsigned 64-bit a - b has underflow
+#define IS_UINT64_SUB_UNDERFLOW(a, b) (a < b)
+
+// Returns true if unsigned 64-bit a + b has overflow
+#define IS_UINT64_ADD_OVERFLOW(a, b) (a > TR::getMaxUnsigned<TR::Int64>() - b)
```

Comment:
> I am really not a fan of macros

I would tend to agree. They offer no benefit over inline functions which are properly typed.